src/uri/include/sourcemeta/jsontoolkit/uri.h

```diff
@@ -301,6 +307,9 @@ class SOURCEMETA_JSONTOOLKIT_URI_EXPORT URI {
   // points to fragments of it.
   // We keep this as const as this class is immutable
   std::string data;
+
+  mutable std::vector<std::string> components;
```

Comment:
Maybe call this `path` or `path_components`? Otherwise its not very clear at first sight that its about paths?

Reply:
> Assuming you do all parsing in the constructor, then you don't need it here?

Yeah. According what we decide here: https://github.com/sourcemeta/jsontoolkit/pull/810#discussion_r1638331668